### DISCUSSION TITLE
Is Many To Many possible in a Filament form?

### DISCUSSION THREAD

**Discussion by ApolloDevs** [filamentphp/filament#287] [Help]
Looking to have a pre-filled checkbox values based on the pivot table from a many-to-many relationship. 

I suppose a good example of this would be roles from Spatie's roles package, allowing you to edit the users' roles. 

I couldn't find a good way of doing this, it doesn't seem there are any column types native to Filament that can handle something like this. 

Could I have an example if possible? Loving the package and have starred it. :)

**Comment by eugenefvdm** [comment]
Googled many to many and Filament and got here. I'm sure it's possible! Going to look further.

**Reply by danharrin** [reply]
https://filamentphp.com/docs/2.x/admin/resources/getting-started#belongstomany

**Reply by eugenefvdm** [reply]
Thanks so much. The key in my case was to "just try it". It appears one can overthink things coming from Nova where you have to add a class. For my use case:

* Invoices belongsToMany Transactions
* Transactions belongsToMany Invoices

I wanted to see all the transactions on a User resource...

```php
php artisan make:filament-relation-manager InvoiceResource transactions transaction_id
```

After running the above artisan command and then adding the RelationManager in `getRelations` of the user resource it just worked.

**Reply by mailtijmen** [reply]
I don't think this really answers the question right?

Following these steps will show the relations but not a form field to create or edit the relations.

**Comment by kakaye-mkubwa** [comment]
Can someone mutate the many-to-many table before update?

**Comment by mailtijmen** [comment]
If you haven't included --attach when creating the relationship manager manually include the various actions in the table section of the relationship manager as set out here:
https://filamentphp.com/docs/2.x/admin/resources/relation-managers#attaching-and-detaching-records 

Just to be complete, if you do not see the relationship manager table underneath the specific model:
check: withPivot() in the eloquent model, and the getRelations in the modelResource

**Reply by abanghendri** [reply]
can you provide an example to do this?
I have withPivot() for my belongsToMany() in my model, but I have no idea where to put in modelResource

**Reply by mailtijmen** [reply]
```
 public static function getRelations(): array
    {
        return [
                
                RelationManagers\CategoriesRelationManager::class,
            
        ];
    }
```
Also Chat probably gives a better explanation than I can
